addons/mail/static/src/js/views/activity/activity_renderer.js

```diff
@@ -174,6 +209,86 @@ var ActivityRenderer = AbstractRenderer.extend({
         var resID = $(ev.currentTarget).data('res-id');
         this.trigger_up('open_view_form', {resID: resID});
     },
+    /**
+     * @private
+     * @param {MouseEvent} ev
+     */
+    _onEmptyCell: function (ev) {
+        var self = this;
+        ev.preventDefault();
+        var $target = $(ev.currentTarget);
+        this.do_action({
+            type: 'ir.actions.act_window',
+            res_model: 'mail.activity',
+            view_mode: 'form',
+            view_type: 'form',
+            views: [[false, 'form']],
+            target: 'new',
+            context: {
+                default_res_id: $target.data('res-id'),
+                default_res_model: this.state.data.model,
+                default_activity_type_id: $target.data('activity-type-id'),
+            },
+            res_id: false,
+        }, { on_close: function () {
+            self.trigger_up('reload');
+        }});
+    },
+    /**
+     * @private
+     * @param {MouseEvent} ev
+     */
+    _onRecordSelector: function (ev) {
+        var self = this;
+        ev.stopPropagation();
+        var existRecords = _.pluck(this.state.data.res_ids, 'id');
+        this.state.data.domain.push(['id', 'not in', existRecords]);
+        return new dialogs.SelectCreateDialog(this, {
+            res_model: this.state.data.model,
+            domain: this.state.data.domain,
+            title: _.str.sprintf(_t("Search: %s"), this.arch.attrs.string),
+            no_create: !this.state.data.can_create,
+            disable_multiple_selection: true,
+            on_selected: function (record) {
+                var KanbanActivity = field_registry.get('kanban_activity');
+                var widget = new KanbanActivity(self, "activity_ids", self._getkanbanActivityData({}, record[0]), {});
+                widget._openActivityForm(false, widget._reload.bind(self));
+            }
+        }).open();
+    },
+    /**
+     * Rearrange body part of table based on active filter.
+     *
+     * @private
+     * @param {OdooEvent} event
+     */
+     _onTweakActivityRecords: function (event) {
+        var self = this;
+        var data = event.target;
```

Comment:
maybe `ev.currentTarget` instead?

Reply:
Here we don't have 'currentTarget'

Reply:
my bad, I thought it was a mouse event.
I think you should pass data in the `trigger_up`, instead of relying on `target`, in order to reduce coupling.